File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file239.f90
subroutine kernel_3mm(ni, nj, nk, nl, nm, e, a, b, f, c, d, g)
implicit none

DATA_TYPE, dimension(nk, ni) :: a
DATA_TYPE, dimension(nj, nk) :: b
DATA_TYPE, dimension(nm, nj) :: c
DATA_TYPE, dimension(nl, nm) :: d
DATA_TYPE, dimension(nj, ni) :: e
DATA_TYPE, dimension(nl, nj) :: f
DATA_TYPE, dimension(nl, ni) :: g
integer :: ni, nj, nk, nl, nm
integer :: i, j, k

!$pragma scop
! E := A*B
do i = 1, _PB_NI
do j = 1, _PB_NJ
e(j,i) = 0.0
do k = 1, _PB_NK
e(j,i) = e(j,i) + a(k,i) * b(j,k)
end do
end do
end do

! F := C*D
do i = 1, _PB_NJ
do j = 1, _PB_NL
f(j,i) = 0.0
do k = 1, _PB_NM
f(j,i) = f(j,i) + c(k,i) * d(j,k)
end do
end do
end do

! G := E*F
do i = 1, _PB_NI
do j = 1, _PB_NL
g(j,i) = 0.0
do k = 1, _PB_NJ
g(j,i) = g(j,i) + e(k,i) * f(j,k)
end do
end do
end do
!$pragma endscop

end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file239.cpp
void kernel_3mm(int ni, int nj, int nk, int nl, int nm, double** e, double** a, double** b, double** f, double** c, double** d, double** g)
{
 int i, j, k;

 // E := A*B
 for (i = 0; i < ni; i++)
 for (j = 0; j < nj; j++)
 {
 e[j][i] = 0.0;
 for (k = 0; k < nk; k++)
 e[j][i] += a[k][i] * b[j][k];
 }

 // F := C*D
 for (i = 0; i < nj; i++)
 for (j = 0; j < nl; j++)
 {
 f[j][i] = 0.0;
 for (k = 0; k < nm; k++)
 f[j][i] += c[k][i] * d[j][k];
 }

 // G := E*F
 for (i = 0; i < ni; i++)
 for (j = 0; j < nl; j++)
 {
 g[j][i] = 0.0;
 for (k = 0; k < nj; k++)
 g[j][i] += e[k][i] * f[j][k];
 }
}
